lib/loader.js

```diff
@@ -100,15 +97,30 @@ module.exports = function loader(content, map) {
 
   plugins.push(icssParser(parserOptions));
 
+  /**
+   * > To ensure that PostCSS generates source maps and displays better syntax
+   * > errors, runners must specify the from and to options. If your runner
+   * > does not handle writing to disk (for example, a gulp transform), you
+   * > should set both options to point to the same file
+   * @see postcss [PostCSS Runner Guidelines]{@link https://github.com/postcss/postcss/blob/master/docs/guidelines/runner.md#21-set-from-and-to-processing-options}
+   *
+   * `css-loader` isn't responsible for writing the map, so it doesn't have to
+   * worry about updating the map with a transformation that changes locations
+   * (suchs as map.file or map.sources).
+   *
+   * Changing the file extension counts as changing the location because it
+   * changes the path.
+   *
+   * PostCSS's `from` and `to` arguments are only concerned with the file
+   * system. They don't know about, care about, or understand the webpack
+   * loader's current request or remaining request.
+   */
+  const { resourcePath } = this;
+
   postcss(plugins)
     .process(content, {
-      // we need a prefix to avoid path rewriting of PostCSS
-      from: `/css-loader!${getRemainingRequest(this)
-        .split('!')
-        .pop()}`,
-      to: getCurrentRequest(this)
-        .split('!')
-        .pop(),
+      from: resourcePath,
```

Comment:
It is dangerous using this, also it is invalid use `resourcePath`, because it is not resource path, it is request, maybe you can create problem repository?

Reply:
It will take me a while to completely respond to this. If I am lucky, I can have something ready some time this week.

I need to understand more about the differences between when to use the request and when to use a file system path.

---

It is difficult to learn the details of the request syntax, because it does not exist on the website (<https://webpack.js.org/concepts/>) nor the documentation used to generate it (<https://github.com/webpack/webpack.js.org>). The closest we come is documentation on context: <https://webpack.js.org/api/loaders/#the-loader-context>.

There are some pages in the old wiki, but they don't go into a lot of detail and users looking for documentation might not expect to look in the deprecated docs repository wiki because they appear to have the same information at first glance and would expect the syntax to be documented on the site. In reality, the wiki contains information not present in the in the more up to date documentation: <https://github.com/webpack/docs/wiki/loaders>, <https://github.com/webpack/docs/wiki/using-loaders>.

---

The next post will probably be a big one. 

Before I make it, I want to see if what I'm concerned about is still an issue or if I'm worried about nothing. My next step is to create the example repository that you requested as it will help illustrate what has been fixed and what still needs work. I can't rely only on the test suite just yet.

Reply:
@npetruzzelli feel free to create minimum reproducible test repo, it is allow to me also identify the problem and how to fix it

Reply:
Hi @evilebottnawi, 

I've created the repository that illustrates the problem. I'll need some time to get some text and images together to support/describe it clearly instead of needing you to dig through multiple pull requests and issues across multiple repositories for a clear picture.

https://github.com/npetruzzelli-forks/webpack-loader-demos 

I plan to use this repository to demonstrate the eventual solution just by pointing to a different Webpack file, probably in a new branch.

Relevant information:
-   OS: **Windows 10 Pro 64-bit**
-   NODE: **v8.9.4**
-   NPM: **6.4.1** (see also: [npm-windows-upgrade](https://github.com/felixrieseberg/npm-windows-upgrade))
-   Windows Build Tools: **[5.0.0](https://github.com/felixrieseberg/windows-build-tools/tree/a1ff38f579f77a205269e9bb6e6f3de5cd0ea2fa)**

Reply:
@npetruzzelli great, thanks, i will see tomorrow, very tired today

Reply:
No rush! I know that feeling. Rest well.

Reply:
Reproducible test repo is not very simple, i try around 30 minutes to understand structure. Also looks many things broken, js source maps also doesn't work in you example. Please simplify example.